control-plane-operator/controllers/hostedcontrolplane/v2/kas/component.go

```diff
@@ -100,7 +102,16 @@ func NewComponent() component.ControlPlaneComponent {
 			component.EnableForPlatform(hyperv1.AWSPlatform),
 			component.WithAdaptFunction(adaptAWSPodIdentityWebhookKubeconfigSecret),
 		).
+		WithManifestAdapter(
+			"azure-kms-secretprovider.yaml",
+			component.WithAdaptFunction(kms.AdaptAzureSecretProvider),
+			component.WithPredicate(isAroHCP),
+		).
 		RolloutOnConfigMapChange("kas-config", "kas-audit-config", "auth-config").
 		RolloutOnSecretChange("kas-secret-encryption-config").
 		Build()
 }
+
+func isAroHCP(cpContext component.WorkloadContext) bool {
+	return azureutil.IsAroHCP()
```

Comment:
I think this predicate should also need to check the HCP spec if Encryption type is KMS